src/csvwidget/widget.ts

```diff
@@ -137,15 +141,21 @@ class CSVWidget extends Widget {
       th.textContent = name;
       header.appendChild(th);
     }
-    for (let row of parsed) {
+    for (let row of parsed.slice(0, RENDER_LIMIT)) {
       let tr = document.createElement('tr');
       for (let col of parsed.columns) {
         let td = document.createElement('td');
-        td.textContent = row[col]
+        td.textContent = row[col];
         tr.appendChild(td);
       }
       body.appendChild(tr);
     }
+    if (parsed.length > RENDER_LIMIT) {
```

Comment:
Let's append the warning below the table so the user will see it on the page if they scroll down all the way and maybe use the word display/show in the message rather than "render"